registration/views/onsite_admin.py

```diff
@@ -211,78 +214,62 @@ def notifyTerminal(request, data):
 
 
 def assignBadgeNumber(request):
-    badge_id = request.GET.get("id")
-    badge_number = request.GET.get("number")
-    badge_name = request.GET.get("badge", None)
-    badge = None
+    msgs = []
+    successful_ids = []
     event = Event.objects.get(default=True)
-    logger.info(
-        "assignBadgeNumber: id='{0}' badge_nembr='{1}' badge_name='{2}'".format(
-            badge_id, badge_number, badge_name
-        )
-    )
-
-    if badge_name is not None:
-        try:
-            badge = Badge.objects.filter(
-                badgeName__icontains=badge_name, event__name=event.name
-            ).first()
-        except BaseException:
-            return JsonResponse(
-                {"success": False, "reason": "Badge name search returned no results"}
-            )
-    else:
-        if badge_id is None or badge_number is None:
-            return JsonResponse(
-                {"success": False, "reason": "id and number are required parameters"},
-                status=400,
-            )
+    queryset = request.POST
 
-    try:
-        badge_number = int(badge_number)
-    except ValueError:
-        return JsonResponse(
-            {"success": False, "message": "Badge number must be an integer"}, status=400
-        )
-    logger.info("assignBadgeNumber: int(badge_number) = {0}".format(badge_number))
-
-    if badge is None:
-        try:
-            badge = Badge.objects.get(id=int(badge_id))
-        except Badge.DoesNotExist:
-            return JsonResponse(
-                {"success": False, "message": "Badge ID specified does not exist"},
-                status=404,
-            )
-        except ValueError:
-            return JsonResponse(
-                {"success": False, "message": "Badge ID must be an integer"}, status=400
-            )
-
-    try:
-        if badge_number < 0:
-            # Auto assign
-            badges = Badge.objects.filter(event=badge.event)
-            highest = badges.aggregate(Max("badgeNumber"))["badgeNumber__max"]
-            highest = highest + 1
-            badge.badgeNumber = highest
-        else:
-            badge.badgeNumber = badge_number
-        badge.save(update_fields=["badgeNumber"])
-        logger.info(
-            "assignBadgeNumber: Badge number saved - {0}".format(badge.badgeNumber)
-        )
-    except Exception as e:
-        return JsonResponse(
-            {
-                "success": False,
-                "message": "Error while saving badge number",
-                "error": str(e),
-            },
-            status=500,
-        )
-
-    return JsonResponse({"success": True})
+    re_badge_id = re.compile('badges\[\d+\]\[id\]')
+    re_ind = re.compile('\[\d+\]')
+    badges = []
+    for key in queryset.keys():
+        if re.match(re_badge_id, key):
+            ind = re.search(re_ind, key).group(0)
+            badge_id = queryset['badges'+ ind +'[id]']
```

Comment:
lol wolf you did this the hard way.

`badge_list = request.POST.getlist("id")`

Reply:
From there you can create a queryset from the ORM `badge_set = Badge.objects.filter(id__in=badge_list)`